plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/util/ElkUtil.java

```diff
@@ -430,11 +430,9 @@ public static void translate(final ElkNode parent, final double xoffset, final d
         for (ElkNode child : parent.getChildren()) {
             // Translate node position
             child.setLocation(child.getX() + xoffset, child.getY() + yoffset);
-            
-            // Translate edges
-            // MIGRATE This used to be outgoing edges; check if it makes sense to translate all contained edges instead
-            child.getContainedEdges().stream().forEach(edge -> translate(edge, xoffset, yoffset));
         }
+        // Translate contained edges
+        parent.getContainedEdges().forEach(edge -> translate(edge, xoffset, yoffset));
```

Comment:
Edges that connect `parent` to its descendants will be translated by this operation, causing the end point at `parent`'s border to be moved. One fix would be to add another `translate(...)` method for edges that has a boolean function as an argument to check, whether an end point at a particular `ElkConnectableShape` should be translated or not.

In fact, that might not even be enough... For edges connected to western ports, for example, the translation should be applied to the corresponding end point's y coordinate, but not to its x coordinate... This is tough.

Please add a discussion to this effect to the code, then I will accept the pull request. ;)